src/rf/apps/workers/process.py

```diff
@@ -58,48 +56,53 @@ def start_polling(self):
                         delete_message = self.check_timeout(record['data'])
                     elif job_type in JOB_VALIDATE:
                         if record['eventSource'] == 'aws:s3':
-                            delete_message = self.validate_image(record)
+                            delete_message = self._validate_image(record)
 
                 if delete_message:
-                    self.q.delete_message(message)
+                    self.queue.remove_message(s3_message)
 
-    def validate_image(self, data):
+    def _validate_image(self, data):
         """
         Use Gdal to verify an image is properly formatted and can be further
         processed.
         data -- attribute data from SQS.
         """
-        # Pass image to Gdal to verify.
-        # If it succeeds continue else return False.
+        def mark_image_valid(s3_uuid):
+            image = LayerImage.objects.get(s3_uuid=s3_uuid)
+            image.status = enums.STATUS_VALID
+            image.save()
+            layer_id = image.layer_id
 
-        key = data['s3']['object']['key']
-        payload_uuid = self.extract_uuid_from_aws_key(key)
+            data = {'s3_uuid': s3_uuid}
+            self.queue.add_message(JOB_REPROJECT, data)
 
-        url = self.make_s3_url(data['s3']['bucket']['name'], key)
+            # Add a message to the queue that we can use to watch for timeouts.
+            data = {
+                'timeout': time.time() + TIMEOUT_SECONDS,
+                'layer_id': layer_id
+            }
+            self.queue.add_message(JOB_TIMEOUT, data, TIMEOUT_DELAY)
```

Comment:
I'm not sure why we need to setup a timeout message for the reproject job. I thought it was just for the EMR job (and the comment in check_timeout seems to indicate this).